File: field/contests/atcoder/abc156/c.py
N = int(input())
X = list(map(int,input().split()))
X.sort()
mi,ma=X[0],X[-1]

ans = 10**18
for P in range(mi,ma+1):
    su = 0
    for j in range(N):
        su += (X[j]-P)**2
    ans = min(ans, su)
print(ans)

File: field/contests/atcoder/abc156/src/bin/c.rs
use proconio::{fastout, input};

#[fastout]
#[allow(non_snake_case, non_upper_case_globals)]
fn main() {
    input! {
        N:i32,
        X:[i32;N]
    }

    let ans = (0..101).map(|p| X.iter().map(|&x_i| (x_i - p).pow(2)).sum::<i32>()).min().unwrap();

    println!("{}",ans);
}
